File: src/types/project.ts
export interface Project {
  id: string;
  name: string;
  path: string;
  ios: {
    bundleId: string;
    version: string;
    buildNumber: number;
    config?: {
      scheme: string;
      configuration: string;
      teamId?: string;
      exportMethod?: 'development' | 'ad-hoc' | 'app-store' | 'enterprise';
      apiKey?: string;
      apiIssuer?: string;
    };
  };
  android: {
    bundleId: string;
    version: string;
    versionCode: number;
    buildCommand?: string;
    config?: {
      firebaseAppId?: string; // Firebase App ID for App Distribution (e.g., 1:1234567890:android:0a1b2c3d4e5f67890)
      distributionGroups?: string; // Comma-separated tester groups (e.g., "qa-team, trusted-testers")
    };
  };
  credentials: {
    iosId?: string;
    androidId?: string;
  };
  notifications?: {
    slack?: {
      webhookUrl: string;
      enabled: boolean;
    };
    discord?: {
      webhookUrl: string;
      enabled: boolean;
    };
    telegram?: {
      botToken: string;
      chatId: string;
      enabled: boolean;
    };
  };
  lastBuild?: BuildHistory;
}

export interface BuildHistory {
  id: string;
  projectId: string;
  platform: 'ios' | 'android';
  version: string;
  buildNumber: number;
  status: 'success' | 'failed' | 'building';
  timestamp: number;
  logs: string;
  logFilePath?: string; // Path to log file (especially for iOS builds)
  releaseNote: string;
  format?: string;
  artifactPath?: string;
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct IosConfig {
    pub scheme: String,
    pub configuration: String,
    pub team_id: Option<String>,
    pub export_method: Option<String>,
    pub api_key: Option<String>,
    pub api_issuer: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct IosPlatform {
    pub bundle_id: String,
    pub version: String,
    pub build_number: u32,
    pub config: Option<IosConfig>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct AndroidPlatform {
    pub bundle_id: String,
    pub version: String,
    pub version_code: u32,
    pub build_command: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct ProjectCredentials {
    pub ios_id: Option<String>,
    pub android_id: Option<String>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct SlackConfig {
    pub webhook_url: String,
    pub enabled: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct DiscordConfig {
    pub webhook_url: String,
    pub enabled: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct TelegramConfig {
    pub bot_token: String,
    pub chat_id: String,
    pub enabled: bool,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct NotificationConfig {
    pub slack: Option<SlackConfig>,
    pub discord: Option<DiscordConfig>,
    pub telegram: Option<TelegramConfig>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct Project {
    pub id: String,
    pub name: String,
    pub path: String,
    pub ios: IosPlatform,
    pub android: AndroidPlatform,
    pub credentials: ProjectCredentials,
    #[serde(default)]
    pub notifications: Option<NotificationConfig>,
}

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct BuildHistory {
    pub id: String,
    pub project_id: String,
    pub platform: String,
    pub version: String,
    pub build_number: u32,
    pub status: String,
    pub timestamp: u64,
    #[serde(default)]
    pub logs: String,
    #[serde(default)]
    pub release_note: String,
    pub format: Option<String>,
    pub artifact_path: Option<String>,
    pub log_file_path: Option<String>,
}
